app/controllers/api/databases.php

```diff
@@ -2885,7 +2885,11 @@ function updateAttribute(
             throw new Exception(Exception::DOCUMENT_INVALID_STRUCTURE, '$id is not allowed for creating new documents, try update instead');
         }
 
-        $database = Authorization::skip(fn () => $dbForProject->getDocument('databases', $databaseId));
+        $database = Authorization::skip(function () use ($queueForUsage, $dbForProject, $databaseId) {
+            $queueForUsage->addMetric(METRIC_DATABASE_API_READ, 1);
+            return $dbForProject->getDocument('databases', $databaseId);
```

Comment:
We need to have 1 count per endpoint, we will reflect all costs inside, we don't need to actually check for every under the hood operation, this will make usage stats extremely unpredictable. 

For example:

createDocument = 1 write, if it has relationships, maybe multiple writes in the metric value
getDocument = 1 read, , if it has relationships, maybe multiple reads in the metric value
listDocuments = 1 read , if it has relationships, maybe multiple reads in the metric value
etc